### DISCUSSION TITLE
[Show and Tell] zkSync Era Vyper Compiler Sandbox Environment

### DISCUSSION THREAD

**Discussion by pcaversaccio** [zkSync-Community-Hub/zksync-developers#460] [Show and Tell]
### Project Name

zkSync Era Vyper Compiler Sandbox Environment

### Repo Link

https://github.com/pcaversaccio/zksync-vyper-sandbox

### Technologies Used

Vyper, Hardhat, zkSync Hardhat Plugins, zkSync Local Development Setup & HARD SKILLS 😄 

### Describe your project. What does it do? Why did you create it? Any interesting challenges you faced?

It's a a sandbox environment for zkSync Era Vyper compiler testing. I created it to check for compiler bugs in the zkSync Vyper compilation pipeline. I found one compiler bug so far here: [First Immutable Variable Initialization](https://github.com/matter-labs/era-compiler-vyper/security/advisories/GHSA-h8jv-969m-94r4).

### What specific feedback are you looking for?

Testing the same contract across different compiler settings & versions is still a PITA. If there is a smart way to do it, please let me know!

### How can the community help or contribute to your project?

Feel free to open PRs that can improve the quality of compiler checks across different compiler settings :).

**Comment by bxpana** [comment]
@pcaversaccio amazing! Thanks for sharing and really appreciate you providing more tools for the Vyper community.

**Reply by pcaversaccio** [reply]
Cheers! For any Vyper enthusiast, I'm the author of snekmate: https://github.com/pcaversaccio/snekmate, an OpenZeppelin-like Vyper library. You can check the `modules` branch to see the latest changes that will be shipped soon via the breaking Vyper release `0.4.0`.
